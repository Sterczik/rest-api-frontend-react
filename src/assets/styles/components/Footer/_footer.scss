.footer {
  background: #3f51b5;
  color: $white;
  &__container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 12px;
  }
  &__list {
    display: flex;
    align-items: center;
  }
  &__item {
    display: inline-block;
    padding: 0;
    width: auto;
  }
  &__link {
    color: $white;
    padding: 8px;
    font-weight: 500;
    text-transform: uppercase;
    border-radius: 3px;
    text-decoration: none;
    position: relative;
    display: block;
  }
}
