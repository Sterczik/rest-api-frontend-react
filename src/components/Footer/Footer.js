import React from 'react';
import { Link } from 'react-router-dom';
import { List, ListItem } from '@material-ui/core';

export const Footer = () => (
  <footer className="footer">
    <div className="container footer__container">
      <div>
        <List className="footer__list">
          <ListItem className="footer__item">
            <Link to="/todos" className="footer__link">Todos</Link>
          </ListItem>
          <ListItem className="footer__item">
            <Link to="/my-account" className="footer__link">Account</Link>
          </ListItem>
          <ListItem className="footer__item">
            <a
              href="https://github.com/Sterczik/pern-rest-starter-client"
              className="footer__link"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </a>
          </ListItem>
        </List>
      </div>
      <div>2018</div>
    </div>
  </footer>
);

export default Footer;
