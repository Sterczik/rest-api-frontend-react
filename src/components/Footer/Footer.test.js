import React from 'react';
import { shallow } from 'enzyme';
import { Footer } from './Footer';

test('should render Footer correctly', () => {
  const wrapper = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
});
