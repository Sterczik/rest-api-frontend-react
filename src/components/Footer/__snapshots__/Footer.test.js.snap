// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container footer__container"
      >
        <div>
          <WithStyles(List)
            className="footer__list"
          >
            <WithStyles(ListItem)
              className="footer__item"
            >
              <Link
                className="footer__link"
                replace={false}
                to="/todos"
              >
                Todos
              </Link>
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              className="footer__item"
            >
              <Link
                className="footer__link"
                replace={false}
                to="/my-account"
              >
                Account
              </Link>
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              className="footer__item"
            >
              <a
                className="footer__link"
                href="https://github.com/Sterczik/pern-rest-starter-client"
                rel="noopener noreferrer"
                target="_blank"
              >
                GitHub
              </a>
            </WithStyles(ListItem)>
          </WithStyles(List)>
        </div>
        <div>
          2018
        </div>
      </div>,
      "className": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(List)
              className="footer__list"
            >
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/todos"
                >
                  Todos
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/my-account"
                >
                  Account
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <a
                  className="footer__link"
                  href="https://github.com/Sterczik/pern-rest-starter-client"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
              </WithStyles(ListItem)>
            </WithStyles(List)>
          </div>,
          <div>
            2018
          </div>,
        ],
        "className": "container footer__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)
              className="footer__list"
            >
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/todos"
                >
                  Todos
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/my-account"
                >
                  Account
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <a
                  className="footer__link"
                  href="https://github.com/Sterczik/pern-rest-starter-client"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
              </WithStyles(ListItem)>
            </WithStyles(List)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/todos"
                  >
                    Todos
                  </Link>
                </WithStyles(ListItem)>,
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/my-account"
                  >
                    Account
                  </Link>
                </WithStyles(ListItem)>,
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <a
                    className="footer__link"
                    href="https://github.com/Sterczik/pern-rest-starter-client"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    GitHub
                  </a>
                </WithStyles(ListItem)>,
              ],
              "className": "footer__list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    className="footer__link"
                    replace={false}
                    to="/todos"
                  >
                    Todos
                  </Link>,
                  "className": "footer__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Todos",
                    "className": "footer__link",
                    "replace": false,
                    "to": "/todos",
                  },
                  "ref": null,
                  "rendered": "Todos",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    className="footer__link"
                    replace={false}
                    to="/my-account"
                  >
                    Account
                  </Link>,
                  "className": "footer__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Account",
                    "className": "footer__link",
                    "replace": false,
                    "to": "/my-account",
                  },
                  "ref": null,
                  "rendered": "Account",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    className="footer__link"
                    href="https://github.com/Sterczik/pern-rest-starter-client"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    GitHub
                  </a>,
                  "className": "footer__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "GitHub",
                    "className": "footer__link",
                    "href": "https://github.com/Sterczik/pern-rest-starter-client",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "GitHub",
                  "type": "a",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018",
          },
          "ref": null,
          "rendered": "2018",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container footer__container"
        >
          <div>
            <WithStyles(List)
              className="footer__list"
            >
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/todos"
                >
                  Todos
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <Link
                  className="footer__link"
                  replace={false}
                  to="/my-account"
                >
                  Account
                </Link>
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                className="footer__item"
              >
                <a
                  className="footer__link"
                  href="https://github.com/Sterczik/pern-rest-starter-client"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
              </WithStyles(ListItem)>
            </WithStyles(List)>
          </div>
          <div>
            2018
          </div>
        </div>,
        "className": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <WithStyles(List)
                className="footer__list"
              >
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/todos"
                  >
                    Todos
                  </Link>
                </WithStyles(ListItem)>
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/my-account"
                  >
                    Account
                  </Link>
                </WithStyles(ListItem)>
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <a
                    className="footer__link"
                    href="https://github.com/Sterczik/pern-rest-starter-client"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    GitHub
                  </a>
                </WithStyles(ListItem)>
              </WithStyles(List)>
            </div>,
            <div>
              2018
            </div>,
          ],
          "className": "container footer__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(List)
                className="footer__list"
              >
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/todos"
                  >
                    Todos
                  </Link>
                </WithStyles(ListItem)>
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <Link
                    className="footer__link"
                    replace={false}
                    to="/my-account"
                  >
                    Account
                  </Link>
                </WithStyles(ListItem)>
                <WithStyles(ListItem)
                  className="footer__item"
                >
                  <a
                    className="footer__link"
                    href="https://github.com/Sterczik/pern-rest-starter-client"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    GitHub
                  </a>
                </WithStyles(ListItem)>
              </WithStyles(List)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(ListItem)
                    className="footer__item"
                  >
                    <Link
                      className="footer__link"
                      replace={false}
                      to="/todos"
                    >
                      Todos
                    </Link>
                  </WithStyles(ListItem)>,
                  <WithStyles(ListItem)
                    className="footer__item"
                  >
                    <Link
                      className="footer__link"
                      replace={false}
                      to="/my-account"
                    >
                      Account
                    </Link>
                  </WithStyles(ListItem)>,
                  <WithStyles(ListItem)
                    className="footer__item"
                  >
                    <a
                      className="footer__link"
                      href="https://github.com/Sterczik/pern-rest-starter-client"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      GitHub
                    </a>
                  </WithStyles(ListItem)>,
                ],
                "className": "footer__list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Link
                      className="footer__link"
                      replace={false}
                      to="/todos"
                    >
                      Todos
                    </Link>,
                    "className": "footer__item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Todos",
                      "className": "footer__link",
                      "replace": false,
                      "to": "/todos",
                    },
                    "ref": null,
                    "rendered": "Todos",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Link
                      className="footer__link"
                      replace={false}
                      to="/my-account"
                    >
                      Account
                    </Link>,
                    "className": "footer__item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Account",
                      "className": "footer__link",
                      "replace": false,
                      "to": "/my-account",
                    },
                    "ref": null,
                    "rendered": "Account",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="footer__link"
                      href="https://github.com/Sterczik/pern-rest-starter-client"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      GitHub
                    </a>,
                    "className": "footer__item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "GitHub",
                      "className": "footer__link",
                      "href": "https://github.com/Sterczik/pern-rest-starter-client",
                      "rel": "noopener noreferrer",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "GitHub",
                    "type": "a",
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018",
            },
            "ref": null,
            "rendered": "2018",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
